src/abundances.jl

```diff
@@ -1,93 +1,77 @@
 # Methods for absolute and relative abundances
 
-struct AbundanceTable{T<:Real} <: AbstractArray{T,2}
-    table::Array{T,2}
-    samples::Vector{S} where S
-    features::Vector{R} where R
-end
-
-function AbundanceTable(df::DataFrame)
-    return AbundanceTable(
-            Matrix(df[:,2:end]),
-            names(df[2:end]),
-            Vector(df[1]))
-end
-function AbundanceTable(table::Array{T,2}) where T<:Real
-    return AbundanceTable(
-             table,
-             Vector{Int64}(1:size(table,2)),
-             Vector{Int64}(1:size(table,1)))
-end
-
-@forward_func AbundanceTable.table Base.getindex, Base.setindex, Base.setindex!, Base.length, Base.size
-
+abundancetable(df::DataFrame) = ComMatrix(Matrix(df[2:end]), string.(df[1]), String.(names(df[2:end])))
+abundancetable(table::AbstractArray{T,2}, site = ["sample_$x" for x in indices(table, 2)],
+    species = ["feature_$x" for x in indices(table, 1)]) where T<:Real =
+    ComMatrix(table, species, site)
 
 """
-Filter an abundance table to the top `n` species accross all samples
+Filter an abundance table to the top `n` features accross all samples
 
-This function also adds a row for "other", which sums the
+This function also adds a row for "other", which sums the abundances of the
+remaining features.
 """
-function filterabund(abun::AbundanceTable, n::Int=10)
+function filterabund(abun::AbstractComMatrix, n::Int=minimum(10, nfeatures(abun)))
     # TODO: add prevalence filter
 
-    totals = [sum(abun[i,:]) for i in 1:size(abun, 1)]
+    totals = featuretotals(abun)
 
     srt = sortperm(totals, rev=true)
 
-    newabun = abun[srt[1:n], :]
+    newabun = getfeature(abun, srt[1:n])
 
-    remainder = [sum(abun[srt[n+1:end], i]) for i in 1:size(abun, 2)]'
+    remainder = [sum(abun.occurrences[srt[n+1:end], i]) for i in 1:size(abun, 2)]'
```

Comment:
In the newest version this should be `occurrences(abun)` rather than `abun.occurrences`, as I think we should try to avoid accessing fields directly.

Reply:
Makes sense, will change.